docs/synapse/userguides/storm_adv_functions.rstorm

```diff
@@ -0,0 +1,708 @@
+      
+      //Attempt to create (or lift) an IPv4 from the input
+      //Return the IPv4 and exit if successful
+      [ inet:ipv4 ?= $ip ]
+      return($node)
+      
+      //Otherwise, atempt to create (or lift) an IPv6 from the input
+      //Return the IPv6 and exit if successful
+      [ inet:ipv6 ?= $ip ]
+      return($node)
+      
+      //If the input is not a valid IPv4 or IPv6, the function
+      // will execute but will not return a node.
+  }
+  
+  //Invoke the function with the specified input and
+  // yield the result (if any) into the pipeline
+  yield $makeIP(8.8.8.8)
+
+
+**Return a node using secondary property deconfliction**
+
+When ingesting or creating `guid-based`_ nodes, a common deconfliction strategy is to check for existing
+nodes using one or more secondary properties (known as `secondary property deconfliction`_). A callable
+function that takes a secondary property value (or values) as input and returns (or creates) the node
+simplifies this process.
+
+::
+  
+  //Create an ou:org node based on an org name (ou:name)
+  
+  //Declare function - takes 'name' as input
+  function genOrgByName(name) {
+      
+      //Check whether input is valid for an ou:name value
+      //If not, return / exit
+      ($ok, $name) = $lib.trycast(ou:name, $name)
+      if (not $ok) { return() }
+      
+      //If name is valid, attempt to identify an existing ou:org
+      //Lift the ou:name node for 'name' (if it exists)
+      // and pivot to an org with that name (if it exists)
+      //Return the existing node if found
+      ou:name=$name -> ou:org
+      return($node)
+      
+      //If an org is not found, create a new ou:org using 'gen' and the name
+      // as input for the org's guid; set the :name prop
+      //Return the new node
+      [ ou:org=(gen, $name) :name=$name ]
+      return($node)
+  }
+  
+  //Invoke the function with input name "The Vertex Project" and yield
+  // the result into the pipeline
+  yield $genOrgByName("The Vertex Project")
+
+
+.. TIP::
+  
+  Synapse includes `gen.*`_ (generator) Storm commands and `$lib.gen`_ APIs that can generate many
+  common guid-based forms using secondary property deconfliction.
+
+
+**Return a value**
+
+Some data sources provide feed-like APIs that allow you to retrieve either the entire feed or just
+retrieve any new items added since your last update. The "last update" time can be stored as :ref:`gloss-node-data`
+on the ``meta:source`` node for the data source. A callable function can retrieve the "last updated" date
+(e.g., to pass the value to another function used to retrieve only the latest feed data).
+
+::
+  
+  function getLastReportDate() {
+      
+      //Invoke an existing function to create (initialize) or retrieve the meta:source node.
+      //The function sets the $srcnode variable to the meta:source node.
+      $initMetaSource()
+      
+      //Yield the meta:source node into this function's pipeline
+      yield $srcnode
```

Comment:
Might be better to just have this be `yield $initMetaSource()` for clarity. We do the module level `$srcnode` for power-up performance to avoid re-initializing the same node repeatedly during ingests but it might be confusing here.